trace/opencensus/trace/ext/django/middleware.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Django middleware helper to capture and trace a request."""
+import logging
+import threading
+
+from opencensus.trace.ext import utils
+from opencensus.trace.ext.django.config import settings
+from opencensus.trace.propagation.google_cloud_format import from_header
+
+_thread_locals = threading.local()
```

Comment:
This is ok for now, but we need to investigate the various concurrency packages.  Some use multiple threads, but others use continuations or similar.  So we'll probably want to look at ContextLocals or similar: http://werkzeug.pocoo.org/docs/0.12/local/